3dview properties (#4938)
qgis/QGIS-Documentation: source/docs/user_manual/working_with_vector/vector_properties.rst

See :ref:`data_defined_labeling` for more information.

.. index:: 3d view properties
.. _`sec_3_d_view`:

> |3d| 3D View Properties

Author: @DelazJ yes we do have icons

> |3d| 3D View Properties

Reviewer: I meant there's no icon in the section title (https://docs.qgis.org/testing/en/docs/user_manual/working_with_vector/vector_properties.html)

=======================

|3d| The :guilabel:`3D View` tab provides settings for vector layers that should be
depicted in the :ref:`label_3dmapview` tool. There are settings available for point, line
and polygon layers. We provide information for the |singleSymbol|:guilabel:`Single symbol`
rendering here.

Point Layers
------------
	

> * You can define different simple 3D shapes like :guilabel:`Sphere`, :guilabel:`Cylinder´,

Reviewer 2: ```suggestion
* You can define different simple 3D shapes like :guilabel:`Sphere`, :guilabel:`Cylinder`,
```

* You can define a :guilabel:`Scale factor` for the 3D shape as well as a
  :guilabel:`Rotation` around the x-, y- and z-axis.

Line layers
-----------
	
* Beneath the :guilabel:`Width`and :guilabel:`Height` settings you can
  define the :guilabel:`Extrusion` of the vector lines. If the lines do not have
  z-values, this makes it is possible to define the height of the lines in 3D.
* With the :guilabel:`Altitude clamping` you define the position of the 
  3D lines relative to the underlying terrain surface, if you have included
  raster elevation data or other 3D vectors. We recommend to use :guilabel:`Absolute` here.
* The :guilabel:`Altitude binding` defines how the feature is clamped to the
  terrain. Either every :guilabel:`Vertex` of the feature will be clamped
  to the terrain or this will be done by the :guilabel:`Centroid`.
* It is possible to |checkbox|:guilabel:`Render as simple 3D lines`.
* Also for lines, the shading can be defined in the menus :guilabel:`Diffuse`, :guilabel:`Ambient`,
  :guilabel:`Specular` and :guilabel:`Shininess`.

Polygon Layers
--------------
	
* As for the other ones, :guilabel:`Height` can be defined in CRS units.

> * Again, :guilabel:`Extrusion` is possible for missing z-values.

Author: @saberraz can we go into detail here. When do we need to set the Extrusion parameter.Thanks(-:

* The :guilabel:`Altitude clamping`, :guilabel:`Altitude binding` can be defined
  as explained above.
* There is an additional option to |checkbox|:guilabel:`Add back faces`
  and |checkbox|:guilabel:`Invert normals`.
* You can define |checkbox|:guilabel:`Edges` by :guilabel:`Width` and :guilabel:`Color`.

Application example
-------------------

To go through the settings explained above you can have a look at the qgis sample data

> and open the project :file:`helsinki.qgz`.

Author: @DelazJ my idea would be to extend the qgis sample dataset with 3d vectors files and a raster with elevation data from the Helsinki dataset I downloaded from Sabers Dropbox. The data are very suitable for setting up a 3d view.

> and open the project :file:`helsinki.qgz`.

Reviewer: I agree that having 3d data in the sample dataset will be helpful when testing 3D features, describing them and screen shooting. An already set project file in the docs repository would also be of great help.
My concern relates to adding the project file to the sample dataset, hence accessible to any user. Such a project could get outdated in few releases and no one would notice. I think that explaining settings (and providing data) should be enough to help anyone to check how features do work, in their own project.

> and open the project :file:`helsinki.qgz`.

Author: @DelazJ there is a link to a ed scene now. Can somebody replace and insert the screenshot I uploaded here? I would prefer a screenshot from ubuntu in this section, which I'm not using. Thanks a lot !

> and open the project :file:`helsinki.qgz`.

Reviewer 2: > @DelazJ there is a link to a ed scene now. Can somebody replace and insert the screenshot I uploaded here? I would prefer a screenshot from ubuntu in this section, which I'm not using. Thanks a lot !

![3d_view_properties](https://user-images.githubusercontent.com/1598812/77001175-cb2dbe80-6959-11ea-9254-326e24a56695.png)

See :ref:`data_defined_labeling` for more information.

.. index:: 3d view properties
.. _`sec_3_d_view`:

|3d| 3D View Properties
=======================

|3d| The :guilabel:`3D View` tab provides settings for vector layers that should be
depicted in the :ref:`label_3dmapview` tool. There are settings available for point, line
and polygon layers. We provide information for the |singleSymbol|:guilabel:`Single symbol`
rendering here.

Point Layers
------------
	

> 	* You can define different simple 3D shapes like :guilabel:`Sphere`, :guilabel:`Cylinder´,

Reviewer 2: Travis is complaining here. You should avoid tabs, and make sure that the indentation is consistent. There are several more of these indentation "errors" further on in the text.

          :guilabel:`Cube`, :guilabel:`Cone`, :guilabel:`Plane` and :guilabel:`Torus`
          defined by their :guilabel:`Radius`, :guilabel:`Size` or :guilabel:`Length´.
	  The unit of size of the 3D shapes refers to the CRS of the project.
	* The shading of the 3D shapes can be defined by the menus :guilabel:`Diffuse`, 
	  :guilabel:`Ambient`, :guilabel:`Specular` and :guilabel:`Shininess`
          (see https://en.wikipedia.org/wiki/Phong_reflection_model#Description)

> 	* If you choose a :guilabel:`3D Model` a whole model is depicted at the position

Reviewer 2: ```suggestion
	* If you choose :guilabel:`3D Model`, their location will be determined
	  by the points in the layer.
```

>           of single point coordinates.

Reviewer 2: ```suggestion
```

> 	* For e.g. visualizing point clouds you can use :guilabel:`Billboard` Shapes

Reviewer 2: ```suggestion
	* For visualizing 3D point clouds you can use :guilabel:`Billboard` Shapes
```

          defined by the :guilabel:`Billboard Height`, :guilabel:`Billboard symbol`and
          :guilabel:`Altitude clamping`. The symbol will have a stable size.

> 	* The :guilabel:`Altitude clamping` by :guilabel:`Absolute`, :guilabel:`Relative`

Reviewer 2: ```suggestion
	*  :guilabel:`Altitude clamping` can be set to :guilabel:`Absolute`, :guilabel:`Relative`
```

> 	  or :guilabel:`Terrain`. These settings refer to the 3D data you also

Reviewer 2: ```suggestion
	  or :guilabel:`Terrain`. These settings refer to the terrain surface defined by the 3D data you
```

> 	  loaded into the 3D view (raster elevation data or 3D vector data) and 

Reviewer 2: ```suggestion
	  loaded into the 3D view (raster elevation data or 3D vector data), and 
```

> 	  define the location for the anchor point of the 3D shape.

Reviewer 2: This is not easy to explain what this means, but perhaps something like

```suggestion
	  define how z/elevation values are interpreted with respect to the underlying terrain surface.
```

> 	* The :guilabel:`Translation` can be used to move objects in x, y and z axis.

Reviewer 2: ```suggestion
	* :guilabel:`Translation` can be used to move objects in x, y and z axis.
```

	* You can define a :guilabel:`Scale factor` for the 3D shape as well as a
	  :guilabel:`Rotation` around the x-, y- and z-axis.

Line layers
-----------
	
	* Beneath the :guilabel:`Width`and :guilabel:`Height` settings you can

> 	  define the :guilabel:`Extrusion` of the vector lines. If there are no

Reviewer 2: ```suggestion
	  define the :guilabel:`Extrusion` of the vector lines. If the lines do not have
```

> 	  z-values this way it is possible to define the height of the vectors in

Reviewer 2: ```suggestion
	  z-values, this makes it is possible to define the height of the lines in
```

          3D.
	* With the :guilabel:`Altitude clamping` you define the position of the 

> 	  3D lines on a surface, given that you inserted raster elevation data or

Reviewer 2: ```suggestion
	  3D lines relative to the underlying terrain surface, if you have included raster elevation data or
```

          other 3D vectors. We recommend to use :guilabel:`Absolute` here.
	* The :guilabel:`Altitude binding` defines how the feature is clamped to the
	  terrain. Either every :guilabel:`Vertex` of the feature will be clamped
          to the terrain or this will be done by the :guilabel:`Centroid`.
	* It is possible to |checkbox|:guilabel:`Render as simple 3D lines`.

> 	* Also here the shading of the 3D lines can be defined by the menus

Reviewer 2: ```suggestion
	* Also for lines, the shading can be defined in the menus
```

          :guilabel:`Diffuse`, :guilabel:`Ambient`, :guilabel:`Specular` and
          :guilabel:`Shininess`.

Polygon Layers
--------------
	

> 	* Here, also the :guilabel:`Height` can be defined in CRS units.

Reviewer 2: ```suggestion
	* As for the other ones, :guilabel:`Height` can be defined in CRS units.
```
